File: acwing/蓝桥杯每日一题2025/week3/5526.平衡细菌/5526.平衡细菌.cpp
#include <iostream>
#include <cstring>
#include <algorithm>
#include <cstdio>
#include <cmath>

using namespace std;

const int N = 2 * 1e5 + 10;
long long a[N], d[N], dd[N];

int main()
{

    int n;
    cin >> n;
    for (int i = 1; i <= n; i++)
        cin >> a[i];
    for (int i = 1; i <= n; i++)
        d[i] = a[i] - a[i - 1];
    for (int i = 1; i <= n; i++)
        dd[i] = d[i] - d[i - 1];
    long long res = 0;
    for (int i = 1; i <= n; i++)
        res += abs(dd[i]);
    printf("%d", res);
    return 0;
}

File: acwing/蓝桥杯每日一题2025/week3/5526.平衡细菌/5526.平衡细菌.py
N = int(2 * 1e5) + 10
d = [0] * N
dd = [0] * N

n = int(input())

a = [0] + list(map(int, input().split()))


for i in range(1, n + 1):
    d[i] = a[i] - a[i - 1]

for i in range(1, n + 1):
    dd[i] = d[i] - d[i - 1]

res = 0
for i in range(1, n + 1):
    res += abs(dd[i])

print(res)
